Sources/RocketBasicComponents/SelectionControls/CheckBoxButton/Component/CheckBoxButton.swift

```diff
@@ -0,0 +1,142 @@
+        
+        let externalBorderLayer = CAShapeLayer()
+        externalBorderLayer.path = UIBezierPath(roundedRect: bounds, cornerRadius: cornerRadiusValue).cgPath
+        externalBorderLayer.fillColor = UIColor.clear.cgColor
+        externalBorderLayer.strokeColor = currentStyle?.unselectedColor?.color.cgColor
+        externalBorderLayer.lineWidth = 10.0 * proportion
+        layer.addSublayer(externalBorderLayer)
+        
+        if isSelected {
+            guard let currentStyle = currentStyle else { return }
+            
+            let internalBorderLayer = CAShapeLayer()
+            internalBorderLayer.path = UIBezierPath(roundedRect: bounds,
+                                                    cornerRadius: cornerRadiusValue).cgPath
+            internalBorderLayer.fillColor = currentStyle.selectedColor?.color.cgColor
+            internalBorderLayer.strokeColor = UIColor.clear.cgColor
+            layer.addSublayer(internalBorderLayer)
+            
+            let checkmarkLayer = CAShapeLayer()
+            checkmarkLayer.path = checkmarkBezierPath().cgPath
+            checkmarkLayer.strokeColor = UIColor.white.cgColor
+            checkmarkLayer.lineWidth = 5.0 * proportion
+            checkmarkLayer.fillColor = UIColor.clear.cgColor
+            checkmarkLayer.lineCap = .round
+            checkmarkLayer.lineJoin = .round
+            layer.addSublayer(checkmarkLayer)
+            
+            self.internalLayer = internalBorderLayer
+        }
+        
+        self.borderLayer = externalBorderLayer
+    }
+    
+    private func checkmarkBezierPath() -> UIBezierPath {
+        let checkmark = UIBezierPath()
+        let bounds = self.bounds
+        let xShift: CGFloat = -0.05 * bounds.width
+            
+        let startPoint = CGPoint(x: bounds.minX + 0.3 * bounds.width + xShift,
+                                 y: bounds.midY)
+        let middlePoint = CGPoint(x: bounds.midX + xShift,
+                                  y: bounds.minY + 0.75 * bounds.height)
+        let endPoint = CGPoint(x: bounds.minX + 0.8 * bounds.width + xShift,
+                               y: bounds.minY + 0.2 * bounds.height)
+        
+        checkmark.move(to: startPoint)
+        checkmark.addLine(to: middlePoint)
+        checkmark.addLine(to: endPoint)
+        
+        let center = CGPoint(x: bounds.midX, y: bounds.midY)
+        let rotationAngle: CGFloat = .pi / 18
+        checkmark.apply(CGAffineTransform(translationX: -center.x, y: -center.y))
+        checkmark.apply(CGAffineTransform(rotationAngle: rotationAngle))
+        checkmark.apply(CGAffineTransform(translationX: center.x, y: center.y))
+        
+        return checkmark
+    }
+}
+
+// MARK: - Haptics
+
+private extension CheckBoxButton {
+    func handleTransformAndHaptic() {
+        guard let effect = currentStyle?.effect else { return }
+        
+        if isHighlighted {
+            if let hapticStyle = effect.hapticFeedback {
```

Comment:
AS we can use it in several place, probably it's better to move it to protocol + extension and confirm where required